From cf09822eeeee802a0314d7b0ef8c5a469558a0d5 Mon Sep 17 00:00:00 2001
From: Jeremie Knuesel <jeremie.knusel@sensefly.com>
Date: Wed, 10 Dec 2014 13:12:17 +0100
Subject: [PATCH 3/6] qtglvideosinkbase: don't use variable name 'interface'
 which is a reserved keyword on Windows

Rename 'interface' argument to 'balance_interface' to fix compilation
with MinGW-w64

Compiling GstQtGLVideoSinkBase with MinGW-w64 fails because the class
has a method argument named 'interface', which appears to be a reserved
keyword on Windows.

https://bugzilla.redhat.com/show_bug.cgi?id=980270
https://bugzilla.gnome.org/show_bug.cgi?id=741332
---
 elements/gstqtvideosink/gstqtglvideosinkbase.cpp | 10 +++++-----
 elements/gstqtvideosink/gstqtglvideosinkbase.h   |  2 +-
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/elements/gstqtvideosink/gstqtglvideosinkbase.cpp b/elements/gstqtvideosink/gstqtglvideosinkbase.cpp
index 6849b13..2efdbef 100644
--- a/elements/gstqtvideosink/gstqtglvideosinkbase.cpp
+++ b/elements/gstqtvideosink/gstqtglvideosinkbase.cpp
@@ -122,13 +122,13 @@ void GstQtGLVideoSinkBase::finalize(GObject *object)
 //------------------------------
 
 
-void GstQtGLVideoSinkBase::colorbalance_init(GstColorBalanceInterface *interface, gpointer data)
+void GstQtGLVideoSinkBase::colorbalance_init(GstColorBalanceInterface *balance_interface, gpointer data)
 {
     Q_UNUSED(data);
-    interface->list_channels = GstQtGLVideoSinkBase::colorbalance_list_channels;
-    interface->set_value = GstQtGLVideoSinkBase::colorbalance_set_value;
-    interface->get_value = GstQtGLVideoSinkBase::colorbalance_get_value;
-    interface->get_balance_type = GstQtGLVideoSinkBase::colorbalance_get_balance_type;
+    balance_interface->list_channels = GstQtGLVideoSinkBase::colorbalance_list_channels;
+    balance_interface->set_value = GstQtGLVideoSinkBase::colorbalance_set_value;
+    balance_interface->get_value = GstQtGLVideoSinkBase::colorbalance_get_value;
+    balance_interface->get_balance_type = GstQtGLVideoSinkBase::colorbalance_get_balance_type;
 }
 
 const GList *GstQtGLVideoSinkBase::colorbalance_list_channels(GstColorBalance *balance)
diff --git a/elements/gstqtvideosink/gstqtglvideosinkbase.h b/elements/gstqtvideosink/gstqtglvideosinkbase.h
index 5cf3381..cfe8bb7 100644
--- a/elements/gstqtvideosink/gstqtglvideosinkbase.h
+++ b/elements/gstqtvideosink/gstqtglvideosinkbase.h
@@ -61,7 +61,7 @@ private:
     static void init(GTypeInstance *instance, gpointer g_class);
     static void finalize(GObject *object);
 
-    static void colorbalance_init(GstColorBalanceInterface *interface, gpointer data);
+    static void colorbalance_init(GstColorBalanceInterface *balance_interface, gpointer data);
     static const GList *colorbalance_list_channels(GstColorBalance *balance);
     static void colorbalance_set_value(GstColorBalance *balance,
                                        GstColorBalanceChannel *channel,
-- 
2.5.0

